spec/support/libraryStubData.js

```diff
@@ -47,6 +53,19 @@ var libraryStubData = {
         arrayToString: function (arr) {
             return arr[0] + 'x' + arr[1];
         },
+        stringToArray: function (arr) {
+                var result = []; 
+                var temp = [];
+                var temp2 = [];
+                temp = arr.split(",");
+                for (var i = 0; i < temp.length; i++) {
+                temp2.push(temp[i].replace("x",","));
+                }
+                for (var i = 0; i < temp2.length; i++) {
+                result.push(new Array(Number(temp2[i].split(",")[0]), Number(temp2[i].split(",")[1])));
+                }
+            return result[0];
```

Comment:
if you're only returning the first element, then why use loops above to populate additional elements of the array?
unless I'm missing something I think this entire function could reduced to the following (could add some error checking too, but you don't have it in the original function either):

`stringToArray: function (str) {`
`   var arr = str.split(",")[0].split("x");`
`   return [Number(arr[0]), Number(arr[1])];`
`}`

or using the functional "array.map()"
`stringToArray: function (str) {`
`   return str.split(",")[0].split("x").map(function (elem) {`
`      return Number(elem);`
`  });`
`}`